### PR TITLE
[CI] Wait for Webdav to be ready

### PR DESCRIPTION
**Pull request by troopa81**
It [happens](https://github.com/qgis/QGIS/pull/44832/checks?check_run_id=3428426882) that Webdav external storage test reaches timeout. 

Normally the test only last one second, so one possible reason is that the NGINX docker isn't ready to answer HTTP requests. This PR makes the test to wait the Nginx server to be ready. 

I'm not 100% sure it comes from this. If the tests timeout even with this, I'll have to find something else...

### CODE REVIEW

**Comment by troopa81** [comment]
> it will be safer:

@MorriganR That's right! Done

File: .docker/docker-qgis-test.sh [hunk 1]
```diff
@@ -175,7 +175,29 @@ EOT
...
+  do
+    printf "."
+    sleep 5
+    if [[ $(( COUNT++ )) -eq 40 ]]; then
+      break
+    fi
+  done
+  if [[ ${COUNT} -eq 41 ]]; then
+    echo "Error: WebDAV docker timeout!!!"
+    exit 1
```
Comment (MorriganR) [inline .docker/docker-qgis-test.sh:196]: it might be worth giving other tests a chance, even if WebDAV fails

Comment (troopa81) [reply .docker/docker-qgis-test.sh]: Yes, I was hesitating a lot on this one, but you're probably right. I removed the exit.

**Comment by MorriganR** [comment]
> It [happens](https://github.com/qgis/QGIS/pull/44832/checks?check_run_id=3428426882) that Webdav external storage test reaches timeout.
> 
> Normally the test only last one second, so one possible reason is that the NGINX docker isn't ready to answer HTTP requests. This PR makes the test to wait the Nginx server to be ready.
> 
> I'm not 100% sure it comes from this. If the tests timeout even with this, I'll have to find something else...

5% of test failures
You fall into this trap https://github.com/qgis/QGIS/pull/44827#issuecomment-907607749

**Comment by troopa81** [comment]
> 5% of test failures
You fall into this trap #44827 (comment)

I saw you PRs and wondering if my issues was, somewhat related to yours...

I try to take a look at your last PR and add a comment, but it looks like a tricky one to fix.

**Pull request merged**